pkg/apps/blackduck/components/pvc/v1/pvc.go

```diff
@@ -0,0 +1,81 @@
+package v1
+
+import (
+	horizonapi "github.com/blackducksoftware/horizon/pkg/api"
+	"github.com/blackducksoftware/horizon/pkg/components"
+	blackduckapi "github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/components/utils"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/types"
+	"github.com/blackducksoftware/synopsys-operator/pkg/apps/store"
+	"github.com/blackducksoftware/synopsys-operator/pkg/protoform"
+	"github.com/blackducksoftware/synopsys-operator/pkg/util"
+	"k8s.io/client-go/kubernetes"
+)
+
+type BdPVC struct {
+	config     *protoform.Config
+	kubeClient *kubernetes.Clientset
+	blackduck  *blackduckapi.Blackduck
+}
+
+func NewPvc(config *protoform.Config, kubeClient *kubernetes.Clientset, blackduck *blackduckapi.Blackduck) types.PVCInterface {
+	return &BdPVC{config: config, kubeClient: kubeClient, blackduck: blackduck}
+}
+
+func (b BdPVC) GetPVCs() ([]*components.PersistentVolumeClaim, error) {
+	var pvcs []*components.PersistentVolumeClaim
+
+	defaultPVC := map[string]string{
+		"blackduck-postgres":         "150Gi",
+		"blackduck-authentication":   "2Gi",
+		"blackduck-cfssl":            "2Gi",
+		"blackduck-registration":     "2Gi",
+		"blackduck-solr":             "2Gi",
+		"blackduck-webapp":           "2Gi",
+		"blackduck-logstash":         "20Gi",
+		"blackduck-zookeeper":        "4Gi",
+		"blackduck-uploadcache-data": "100Gi",
+	}
+
+	if b.blackduck.Spec.ExternalPostgres != nil {
+		delete(defaultPVC, "blackduck-postgres")
```

Comment:
Rather than delete, add if ExternalPostgres == nil.  Only have defaultPVC initial definition be ones we always want.